Sources/_StringProcessing/Engine/MEBuiltins.swift

```diff
@@ -15,7 +15,7 @@ extension Processor {
     isStrictASCII: Bool,
     isScalarSemantics: Bool
   ) -> Bool {
-    guard let next = input.matchBuiltinCC(
+    guard currentPosition < end, let next = input.matchBuiltinCC(
```

Comment:
Do you think it would make more sense for `matchBuiltinCC` to take a limited by parameter? What should happen if the substring splits a grapheme cluster like `/d` for `1\u{301}` where the substring doesn't have the combining accent?

Reply:
You're right — I think having a `limitedBy` parameter is the right behavior, since it will let us address split-character indexes in a way that just having a precondition won't.